### PR TITLE
feat(dispatcher): support keep_throttled_payloads and keep_throttled_documents

### PR DESCRIPTION
**Pull request by terencewu-shopline**

### CODE REVIEW

File: internal/outgoing/dispatcher.go [hunk 1]
```diff
@@ -161,23 +189,45 @@ func (d *Dispatcher) processDelayedEvent(e models.IncomingEvent, timeRemain time
...
+	}
+
+	if dest.Config.IncludeDocument && dest.Config.KeepThrottledDocuments {
+		customizedDocument := d.customizeDocument(&e, dest, documentStore)
+		queueKey := getEventThrottledDocumentsKey(store, e, dest)
+		jsonString, jsonErr := json.Marshal(customizedDocument)
+		if jsonErr != nil {
+			panic(jsonErr)
+		}
+		store.Enqueue(queueKey, string(jsonString), dest.Config.GetThrottleValue()*2)
```
Comment (tinyjacky) [inline internal/outgoing/dispatcher.go:225]: does that mean if i enabled keep throttled docs & throttle payload, the event would be enqueued twice?

Comment (tinyjacky) [reply internal/outgoing/dispatcher.go:225]: oh nvm, looks like i'm confused by the word "enqueue" as they are actually 2 diff redis key just keeping the payloads and docs separately

Comment (terencewu-shopline) [reply internal/outgoing/dispatcher.go:225]: @tinyjacky `jsonString` here and above are different.
one is the `payload`, another is the `document`
I can change the variable name to be more specific.

**Review by tinyjacky — APPROVED** [review]

File: internal/outgoing/dispatcher.go (hunk 1 above)
Comment (tinyjacky) [reply internal/outgoing/dispatcher.go:225]: Yes I am now able to notice that. I think you can keep it as is

**Review by Seitk — APPROVED** [review]

**Pull request merged**